extensions/workspace/watch/watch.cmd.tsx

```diff
@@ -1,8 +1,39 @@
 import { Command, CommandOptions } from '@teambit/cli';
 
 import { Watcher } from './watcher';
+import chalk from 'chalk';
+import moment from 'moment';
+import logger from 'bit-bin/dist/logger/logger';
+import { formatCompileResults, formatWatchPathsSortByComponent } from './output-formatter';
 
 export class WatchCommand implements Command {
+
+  msgs = {
```

Comment:
you should make this the opposite.
Instead of the watcher calling the msgs from the cmd.
the watcher should provide a way to register to his events and the cmd should register to his events

Reply:
This is exactly whats happened :)
`await this.watcher.watch({ msgs: this.msgs, verbose });`

Reply:
it's not exactly the same, currently, the logic is aware of a specific subscriber (the command) and call it directly. 
it should provide a way for other to register to the events without knowing who registers and who doesn't.
but it's not critical for now.